doc/source/biascorr.md

```diff
@@ -15,146 +17,420 @@ kernelspec:
-deramp.fit(ref_dem, tba_dem, inlier_mask=inlier_mask)
+```{code-cell} ipython3
+# Instantiate a 2nd order 2D deramping
+deramp = xdem.coreg.Deramp(poly_order=2)
+# Fit and apply
+corrected_dem = deramp.fit_and_apply(ref_dem, tbc_dem_ramp)
+```
 
-# Apply the transformation
-corrected_dem = deramp.apply(tba_dem)
+```{code-cell} ipython3
+:tags: [hide-input]
+:mystnb:
+:  code_prompt_show: "Show plotting code"
+:  code_prompt_hide: "Hide plotting code"
+
+# Plot before and after
+f, ax = plt.subplots(1, 2)
+ax[0].set_title("Before deramp")
+(tbc_dem_ramp - ref_dem).plot(cmap='RdYlBu', ax=ax[0])
+ax[1].set_title("After deramp")
+(corrected_dem - ref_dem).plot(cmap='RdYlBu', vmin=-30, vmax=30, ax=ax[1], cbar_title="Elevation differences (m)")
+_ = ax[1].set_yticklabels([])
 ```
 
-## Directional biases
+### Directional biases
 
 {class}`xdem.coreg.DirectionalBias`
 
-- **Performs:** Correct biases along a direction of the DEM.
-- **Supports weights** Yes.
-- **Recommended for:** Undulations or jitter, common in both stereo and radar DEMs.
+- **Performs:** Correct biases along a direction.
+- **Supports weights:** Yes.
+- **Pros:** Correcting undulations or jitter, common in both stereo and radar DEMs, or strips common in scanned imagery.
+- **Cons:** Long optimization for a sum of sinusoids.
 
-The default optimizer for directional biases optimizes a sum of sinusoids using 1 to 3 different frequencies, and keeping the best performing fit.
+The default optimizer for directional biases fits a sum of sinusoids using 1 to 3 different frequencies and
+keeps the best performing fit, which is useful for periodic along-track errors common to DEMs.
 
-### Example
+```{code-cell} ipython3
+:tags: [hide-cell]
+:mystnb:
+:  code_prompt_show: "Show the code for adding a sum of sinusoids bias"
+:  code_prompt_hide: "Hide the code for adding a sum of sinusoids bias"
+
+# Get rotated coordinates along an angle
+angle = 20
+xx = gu.raster.get_xy_rotated(ref_dem, along_track_angle=angle)[0]
+
+# One sinusoid: amplitude, phases and frequencies
+params = np.array([(15, 10000, np.pi)]).flatten()
+
+# Create a sinusoidal bias and add to the DEM
+from xdem.fit import sumsin_1d
+synthetic_bias = sumsin_1d(xx.flatten(), *params).reshape(np.shape(ref_dem.data))
+tbc_dem_sumsin = ref_dem + synthetic_bias
+```
+
+
+```{code-cell} ipython3
+# Define a directional bias correction at a certain angle (degrees), defaults to "bin_and_fit" for a sum of sinusoids
+dirbias = xdem.coreg.DirectionalBias(angle=20)
+# Fit and apply
+corrected_dem = dirbias.fit_and_apply(ref_dem, tbc_dem_sumsin, random_state=42)
+```
+
+```{code-cell} ipython3
+:tags: [hide-input]
+:mystnb:
+:  code_prompt_show: "Show plotting code"
+:  code_prompt_hide: "Hide plotting code"
+
+# Plot before and after
+f, ax = plt.subplots(1, 2)
+ax[0].set_title("Before directional\nde-biasing")
+(tbc_dem_sumsin - ref_dem).plot(cmap='RdYlBu', vmin=-30, vmax=30, ax=ax[0])
+ax[1].set_title("After directional\nde-biasing")
```

Comment:
The example plotted in the doc is not working... The bias is still very visible.

Reply:
Actually, the bias seem to be doubled.

Reply:
Yep, this is the ongoing issue described in the PR comment above that I can't seem to fix (it either corrects perfectly, or not at all... with the same `random_state`). :cry: